# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/base/pickle.h
# Commit: c02fa459e77d
# Full Hash: c02fa459e77d2572b2fe740f20d6070a77383589
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 72dc6537cf0b -r c02fa459e77d ipc/chromium/src/base/pickle.h
--- a/ipc/chromium/src/base/pickle.h	Mon Jun 21 21:53:11 2021 +0000
+++ b/ipc/chromium/src/base/pickle.h	Mon Jun 21 21:53:11 2021 +0000
@@ -154,6 +154,10 @@
   // will succeed.
   bool HasBytesAvailable(const PickleIterator* iter, uint32_t len) const;
 
+  // Truncate the message at the current point, discarding any data after this
+  // point in the message.
+  void Truncate(PickleIterator* iter);
+
   // Methods for adding to the payload of the Pickle.  These values are
   // appended to the end of the Pickle's payload.  When reading values from a
   // Pickle, it is important to read them in the order in which they were added